docs/src/miden-node/miden-node-block-producer.md

```diff
@@ -0,0 +1,24 @@
+# Miden block producer
+The **Block producer** receives transactions from the RPC component, processes them, creates block containing those transactions before sending created blocks to the store. 
+
+**Block Producer** is one of components of the Miden node. 
+
+## Architecture
+`TODO`
+
+## API
+The **Block Producer** serves connections using the [gRPC protocol](https://grpc.io) on a port, set in the previously mentioned configuration file. The API cannot directly be called by the Miden Client. 
+
+Here is a brief description of supported methods.
+
+### SubmitProvenTransaction
```

Comment:
Do we need to add documentation of these methods here? That means that changes to the API on `miden-node` need to update the docs here on `miden-objects` too. Which is easy to forget and end up with wrong docs.

Reply:
One idea would be to move the method documentation to the `.proto` file themselves. That is then included in the generated rust code, for example:

https://github.com/0xPolygonMiden/miden-node/blob/58be2733cae265c8937391acc72cdc52d98af117/proto/src/generated/account.rs#L4-L11

And then we can just link to the `docs.rs`

Reply:
I agree with @hackaugusto that this is probably not the best place for the API docs as they will quickly get out of sync. I also agree that ideally API docs should be in the protobuf files.

Not sure if exposing protobuf docs via `docs.rs` is the best option. I wonder if there are some specialized tools for building language-independent docs from protobuf files. Maybe we can run it similar to how we run `mdbook` on PR merges?

Reply:
One more general question here would be about the goal of our docs. 

I believe the docs should explain "how to use Miden," which translates to "how to build on Miden." If this is true, the docs should be the primary place for the API docs, right?

Reply:
> Not sure if exposing protobuf docs via docs.rs is the best option. I wonder if there are some specialized tools for building language-independent docs from protobuf files. Maybe we can run it similar to how we run mdbook on PR merges?

I was thinking about the rust generated code, that inherits the documentation from the protos, and then we they can be published as part of the `miden-proto` crate.

Reply:
> I believe the docs should explain "how to use Miden," which translates to "how to build on Miden." If this is true, the docs should be the primary place for the API docs, right?

I'm thinking of these docs provide a higher-level explanation of the system, and should link to more detailed docs (e.g., miden VM, node APi, smart contract SDK). Putting everything into one place would probably be too much.

> I was thinking about the rust generated code, that inherits the documentation from the protos, and then we they can be published as part of the `miden-proto` crate.

Yep - I think this could work as an interim solution, but i was thinking of the case when someone would like to interact with the node using Go, JavaScript etc. It'd feel a bit unnatural to go to the Rust doc site to read language-independent protobuf specs.